docs/migration-guides.md

```diff
@@ -0,0 +1,179 @@
+      <artifactId>kalix-spring-boot-starter</artifactId>
+      <version>${kalix-sdk.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>io.kalix</groupId>
+      <artifactId>kalix-spring-boot-starter-test</artifactId>
+      <version>${kalix-sdk.version}</version>
+      <scope>test</scope>
+    </dependency>
+```
+
+#### Spring Boot Configurations
+
+The following configurations are no longer needed and should be removed: 
+- `spring.main.web-application-type=none` in `application.properties` (feel free to remove the file if that's the only property there)
+- in your `Main.java`, you can now remove `@Import(KalixConfiguration.class)`
+
+#### EventSourcedEntity with typed events
+
+A `EventSourcedEntity<S>` is now `EventSourcedEntity<S, E>` where `E` is the top type for all the different event types such event-sourced entity will be allowed to emit. 
+
+Thus, instead of :
+```java
+public class ShoppingCartEntity extends EventSourcedEntity<ShoppingCart> {
+  // command and event handlers ...
+}
+```
+There should be now:
+```java
+public class ShoppingCartEntity extends EventSourcedEntity<ShoppingCart, ShoppingCartEvent> {
+  // command and event handlers ...
+}
+```
+
+Where an example of this `ShoppingCartEvent`, could be:
+```java
+public sealed interface Events {
+
+  @TypeName("item-added")
+  record ItemAdded(ShoppingCart.LineItem item) implements Events { }
+
+  @TypeName("item-removed")
+  record ItemRemoved(String productId) implements Events { }
+}
+```
+
+#### Kalix Annotations
+
+All Kalix annotations moved from `kalix.springsdk.annotations.*` to `kalix.javasdk.annotations.*`.
+
+Thus, when importing them, where before there was for instance:
+```java
+import kalix.springsdk.annotations.EntityKey;
+import kalix.springsdk.annotations.EntityType;
+import kalix.springsdk.annotations.EventHandler;
+import kalix.springsdk.annotations.GenerateEntityKey;
+```
+
+It should now be:
+```java
+import kalix.javasdk.annotations.EntityKey;
+import kalix.javasdk.annotations.EntityType;
+import kalix.javasdk.annotations.EventHandler;
+import kalix.javasdk.annotations.GenerateEntityKey;
+```
+
+Note these are only a few example of the existing annotations but all of them were moved.
+
+#### TestKits
+
+The Entity-specific test kits moved from `kalix.springsdk.testkit.*` to `kalix.javasdk.testkit.*`.
+
+Thus, you will need to replace these:
+```java
+import kalix.springsdk.testkit.ActionTestkit;
+import kalix.springsdk.testkit.EventSourcedTestKit;
+import kalix.springsdk.testkit.ValueEntityTestKit;
+```
+
+With:
+
+```java
+import kalix.javasdk.testkit.ActionTestkit;
+import kalix.javasdk.testkit.EventSourcedTestKit;
+import kalix.javasdk.testkit.ValueEntityTestKit;
+```
+
+Lastly, `KalixIntegrationTestKitSupport` was also moved, so you will need to replace:
+```java
+import kalix.springsdk.testkit.KalixIntegrationTestKitSupport;
+```
+With:
+```java
+import kalix.spring.testkit.KalixIntegrationTestKitSupport;
+```
```

Comment:
There is one more thing to change. 

Users should delete `src/main/resources/application.properties`.

This file contains a setting to tell Spring to not start the webserver. That was needed in order to not start the spring netty.  

This now handled by the `starter` and if this old hack in place, our own server (based on akka-http). Our server is now configured as the default Spring reactive server, but the settings tell Spring to not start one if it finds one.

Reply:
That is mentioned in line #98 under "Spring Boot Configurations".. let me know what you think

Reply:
Oh, I missed. Thanks!